src/bindings/python/tests/test_transformations/test_pattern_ops.py

```diff
@@ -108,10 +107,9 @@ def test_optional_half_match():
     pattern_relu = ops.relu(pattern_abs.output(0))
 
     matcher = Matcher(pattern_relu, "FindRelu")
-    assert matcher.match(model_relu1)
+    assert not matcher.match(model_relu1)
 
```

Comment:
Well, I see that this test case has become a stumbling block. From my understanding of Optional, this use case should match because the bottom part of the graph, but if you and Ivan decided this is not a common use case, then let it be that way.

Reply:
removed as not represented. Thanks for confirmation